libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -290,6 +306,44 @@ TEST_F(CppInterface, DisassemblyOption) {
   EXPECT_THAT(result_from_cloned_options.GetData(), HasSubstr("MemoryModel"));
 }
 
+TEST_F(CppInterface, GenerateDebugInfoBinary) {
+  options_.SetGenerateDebugInfo();
+  // The output binary should contain the name of the vector: debug_info_sample as char
+  // array.
+  EXPECT_THAT(CompilationOutput(kMinimalDebugInfoShader,
+                                shaderc_glsl_vertex_shader, options_),
+              HasSubstr("debug_info_sample"));
```

Comment:
We could and should make this check stronger.

I would favour making a utility function that takes a SPIR-V binary, and a name string, and checks for an OpName instruction for an ID for that name. 
You can use the spvBinaryParse function from SPIR-V Tools to do the parsing for you, and send in a lambda for the instruction handler. 

Also add a TODO to check for OpLine annotations.  That can only be added once glslang supports full debug output (including OpLine)